lib/vc.js

```diff
@@ -82,24 +90,54 @@ async function issue(options = {}) {
   });
 
   // run common credential checks
-  await _checkCredential({credential: options.credential});
+  const {credential} = options;
+  if(!credential) {
+    throw new Error('"credential" property is required for issuing.')
+  }
+  _checkCredential(credential);
 
-  return jsigs.sign(options.credential, {purpose, documentLoader, ...options});
+  return jsigs.sign(credential, {purpose, documentLoader, ...options});
 };
 
 /**
- * Verify a credential.
+ * Verifies a credential or presentation.
+ *
+ * @param {object} [options={}]
+ *
+ * Required, one of:
+ * @param {object} [options.credential] - Verifiable credential
+ * @param {object} [options.presentation] - Verifiable Presentation
+
+ * @param {CredentialIssuancePurpose} [options.purpose]
+ * @param {function} [options.documentLoader]
+ *
+ * @returns {Promise<{verified: boolean, results: *[], error: Error}>}
  */
 async function verify(options = {}) {
-  const {credential} = options;
+  const {credential, presentation} = options;
+
   // run common credential checks
   try {
-    await _checkCredential({credential});
-  } catch(e) {
+    if(!credential && !presentation) {
+      throw new Error(
+        'A "credential" or "presentation" property is required for verifying.')
+    }
```

Comment:
This should be thrown as a `TypeError` outside of the try/catch, it's an abuse of the API. @davidlehn has more comments.

Reply:
Thanks! Fixed.